addon/index.js

```diff
@@ -1,28 +1,43 @@
-import Ember from 'ember';
-
-export default function(app, prefix) {
-  var regex = new RegExp('^' + prefix + '\/((?:instance-)?initializers)\/');
-  var getKeys = (Object.keys || Ember.keys);
-  var moduleNames = getKeys(requirejs._eak_seen);
+function resolveInitializer(moduleName) {
+  var module = require(moduleName, null, null, true);
+  if (!module) {
+    throw new Error(moduleName + ' must export an initializer.');
+  }
+  var initializer = module['default'];
+  if (!initializer.name) {
+    initializer.name = moduleName.slice(moduleName.lastIndexOf('/') + 1);
```

Comment:
loader.js currently still caters to IE8 users of ember, this will need to be changed to support those.